modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php

```diff
@@ -113,4 +111,48 @@ public function loadMultipleAgendaItems(array $ids) {
     return $this->termStorage->loadMultiple($ids);
   }
 
+  /**
+   * Creates default agenda for a meeting.
+   *
+   * @param int $meetingId
+   *   The meeting id.
+   * @param string $agendaName
+   *   The agenda's name.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function createDefaultAgenda(int $meetingId, string $agendaName) {
+    $defaultAgenda = $this->termStorage->getQuery()->accessCheck(FALSE)
+      ->condition('vid', 'event_agendas')
+      ->condition('field_event', $meetingId)
+      ->condition('field_is_default_agenda', TRUE)
+      ->execute();
+    $defaultAgenda = reset($defaultAgenda);
+
+    if (!empty($defaultAgenda)) {
+      return;
+    }
+
+    // If there is a non-default agenda with that name make it default.
+    $properties = [
```

Comment:
Existing agenda items should not be changed. Return instead a message to inform the user.